core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -682,6 +690,48 @@ class ReplicaManager(val config: KafkaConfig,
             FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY)
         }
 
+        println("***************************Go inside read*******************************");
+
+        var batchs = logReadInfo.records.batches().iterator()
+        while(batchs.hasNext()){
+          var currentBatch = batchs.next();
+          var records = currentBatch.streamingIterator(decompressionBufferSupplier);
+          while(records.hasNext())
+          {
+            var record = records.next();
+            var valueByteArray = Utils.toArray(record.key());
+            for( a <- 0 to (valueByteArray.length - 1)) {
+              println("********************************** record.key() = %d ************************".format(valueByteArray(a)));
+            }
+          }
+        }
+
+        while (batchs.hasNext()) {
+          println("********************************** batchs.hasNext() ************************");
+          var currentBatch = batchs.next();
+          var records = currentBatch.streamingIterator(decompressionBufferSupplier);
+          while (records.hasNext()) {
+            println("********************************** record.hasNext() ************************");
+            var record = records.next();
+            var valueByteArray = Utils.toArray(record.key());
+            for (a <- 0 to (valueByteArray.length - 1)) {
+              println("********************************** record.key() = %d ************************".format(valueByteArray(a)));
+            }
+
+            var key = new String(valueByteArray, StandardCharsets.UTF_8);
+            println("********************************** record.key() = %s ************************".format(key));
+
+            if (eventPattern != null) {
+              println("***************************event pattern not equal to null*******************************");
+              var eventPatternRegex = eventPattern.r;
+              var records = logReadInfo.records.batches().iterator();
+              if (!eventPatternRegex.findFirstIn(key.toString).isDefined) {
+                records.remove();
```

Comment:
Need we store the final records to the result?

Reply:
Need to double check and confirm this thing after resolve get out key failed issue.